src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js

```diff
@@ -0,0 +1,429 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import {
+  Tbody,
+  Th,
+  Thead,
+  Tr,
+  Td,
+  TableComposable,
+} from '@patternfly/react-table';
+import {
+  SearchInput,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  MenuToggle,
+  Pagination,
+  Popper,
+  EmptyState,
+  EmptyStateIcon,
+  Title,
+  EmptyStateBody,
+  EmptyStatePrimary,
+  Button,
+  Bullseye,
+  ToolbarGroup,
+  ToolbarFilter,
+  PaginationVariant,
+} from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import { FilterIcon } from '@patternfly/react-icons';
+import Loading from '../LoadingState/Loading';
+import Unavailable from '../AdditionalRepositoriesTable/NoAdditionalRepositories';
+
+const EditAdditionalRepositoriesTable = (props) => {
+  const { repositories, isLoading, error } = props;
+
+  const columnNames = {
+    repositoryName: 'Name',
+    repositoryLabel: 'Label',
+  };
+
+  /* Filter options rationale*/
+  const [searchValue, setSearchValue] = React.useState('');
+  const onSearchChange = (value) => {
+    setSearchValue(value);
+  };
+  const onFilter = (repo) => {
+    let searchValueInput;
+    try {
+      searchValueInput = new RegExp(searchValue, 'i');
+    } catch (err) {
+      searchValueInput = new RegExp(
+        searchValue.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'),
+        'i'
+      );
+    }
+    const matchesSearchValue =
+      repo.repositoryName.search(searchValueInput) >= 0;
+    const matchesLabelValue =
+      repo.repositoryLabel.search(searchValueInput) >= 0;
+    return (
+      (searchValue === '' || matchesSearchValue) &&
+      (searchValue === '' || matchesLabelValue)
+    );
+  };
+  const filteredRepos = repositories?.filter(onFilter);
```

Comment:
I don't think this search filter is working as intended. I would maybe check how it's done in subscription inventory, but the search wasn't filtering correctly when using the edit modal.  

here's how we do it with product table, but the logic will need to change a little:

  const filterDataBySearchTerm = (data: Product[], searchValue: string): Product[] => {
    return data.filter((entry: Product) => {
      const searchTerm = searchValue.toLowerCase().trim();
      const name = (entry.name || '').toLowerCase();
      const productLine = (entry.productLine || '').toLowerCase();
      const sku = (entry.sku || '').toLowerCase();
      return (
        name.includes(searchTerm) || productLine.includes(searchTerm) || sku.includes(searchTerm)
      );
    });
  };